core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java

```diff
@@ -245,7 +245,9 @@ public Plan plan(Analysis analysis, Stage stage, boolean collectPlanStatistics)
         if (stage.ordinal() >= OPTIMIZED.ordinal()) {
             for (PlanOptimizer optimizer : planOptimizers) {
                 root = optimizer.optimize(root, session, symbolAllocator.getTypes(), symbolAllocator, idAllocator, warningCollector, tableStatsProvider);
-                requireNonNull(root, format("%s returned a null plan", optimizer.getClass().getName()));
+                if (root == null) {
+                    throw new NullPointerException(optimizer.getClass().getName() + " returned a null plan");
+                }
```

Comment:
I doubt this is significant enough to warrant making the code more verbose and harder to read.

Reply:
While I'm sure that the performance cost isn't _massive_, this is called inside of the loop for each optimizer. On option would be to switch from `requireNonNull(root, format("%s ..."))` to `requireNonNull(root, () -> format("%s ..."))` but that would still be a capturing lambda creation per loop iteration even if it avoids the `String.format` overhead, which is why I changed it over to an explicit null check.

Reply:
True, but there are only a handful of optimizers. Also, they do way more complicated things, so this is going to be a drop in the bucket.

Reply:
Does lambda capture matter here? I suspect this would be optimized away by inlining.

Reply:
It won't matter as long as the JIT does actually inline the `requireNonNull` _and_ the lambda, but if either one of those doesn't happen then you'll get an extra allocation inside of the loop (which would still be better than calling `String.format` inside of each loop iteration, but is avoidable by using an explicit null check).